core/app/models/spree/credit_card.rb

```diff
@@ -46,12 +46,15 @@ def self.default
 
     def default
       Spree::Deprecation.warn("CreditCard.default is deprecated. Please use user.wallet.default_wallet_payment_source instead.", caller)
-      user.wallet.default_wallet_payment_source.payment_source == self
+      return false if user.nil?
+      user.wallet.default_wallet_payment_source.try!(:payment_source) == self
     end
 
     def default=(set_as_default)
       Spree::Deprecation.warn("CreditCard.default= is deprecated. Please use user.wallet.default_wallet_payment_source= instead.", caller)
-      if set_as_default # setting this card as default
+      if user.nil?
```

Comment:
It looks like our permitted attrs prevent setting credit card defaults via the api.
The admin doesn't allow setting the `default` attribute either.

I'm not seeing any other code that tries to set the default on a credit card, other than the above items.